src/thread_task.rs

```diff
@@ -802,160 +803,149 @@ pub unsafe extern "C" fn rav1d_worker_task(data: *mut c_void) -> *mut c_void {
-                            || (*t).type_0 as c_uint
+                        if t.type_0 as c_uint == RAV1D_TASK_TYPE_TILE_ENTROPY as c_int as c_uint
+                            || t.type_0 as c_uint
                                 == RAV1D_TASK_TYPE_TILE_RECONSTRUCTION as c_int as c_uint
                         {
                             // if not bottom sbrow of tile, this task will be re-added
                             // after it's finished
                             if check_tile(t, f, ((*c).n_fc > 1 as c_uint) as c_int) == 0 {
-                                found = true;
-                                break 'decoding;
+                                break 'found (f, t, prev_t);
                             }
-                        } else if (*t).recon_progress != 0 {
-                            let p = ((*t).type_0 as c_uint
+                        } else if t.recon_progress != 0 {
+                            let p = (t.type_0 as c_uint
                                 == RAV1D_TASK_TYPE_ENTROPY_PROGRESS as c_int as c_uint)
                                 as c_int;
-                            let error = (*f).task_thread.error.load(Ordering::SeqCst);
-                            if !((*f).task_thread.done[p as usize].load(Ordering::SeqCst) == 0
+                            let error = f.task_thread.error.load(Ordering::SeqCst);
+                            if !(f.task_thread.done[p as usize].load(Ordering::SeqCst) == 0
                                 || error != 0)
                             {
                                 unreachable!();
                             }
-                            let frame_hdr = &***(*f).frame_hdr.as_ref().unwrap();
+                            let frame_hdr = &***f.frame_hdr.as_ref().unwrap();
                             let tile_row_base =
-                                frame_hdr.tiling.cols * (*f).frame_thread.next_tile_row[p as usize];
+                                frame_hdr.tiling.cols * f.frame_thread.next_tile_row[p as usize];
                             if p != 0 {
-                                let p1_0 =
-                                    (*f).frame_thread.entropy_progress.load(Ordering::SeqCst);
-                                if p1_0 < (*t).sby {
+                                let p1_0 = f.frame_thread.entropy_progress.load(Ordering::SeqCst);
+                                if p1_0 < t.sby {
                                     break 'next;
                                 }
-                                (*f).task_thread
+                                f.task_thread
                                     .error
                                     .fetch_or((p1_0 == TILE_ERROR) as c_int, Ordering::SeqCst);
                             }
-                            let frame_hdr = &***(*f).frame_hdr.as_ref().unwrap();
+                            let frame_hdr = &***f.frame_hdr.as_ref().unwrap();
```

Comment:
```suggestion
```

This one seems unnecessary; the one above is still in scope.  I understand for the others, though, that they can't be hoisted because of `&mut f`s.